server/verify.go

```diff
@@ -22,14 +25,21 @@ type VerifyOpts struct {
 	// The nonce used when calling client.Attest
 	Nonce []byte
 	// Trusted public keys that can be used to directly verify the key used for
-	// attestation. This option should be used if you already know the AK.
+	// attestation. This option should be used if you already know the AK, as
+	// it provides the highest level of assurance.
 	TrustedAKs []crypto.PublicKey
 	// Allow attestations to be verified using SHA-1. This defaults to false
 	// because SHA-1 is a weak hash algorithm with known collision attacks.
 	// However, setting this to true may be necessary if the client only
 	// supports the legacy event log format. This is the case on older Linux
 	// distributions (such as Debian 10).
 	AllowSHA1 bool
+	// A collection of trusted root CAs that are used to sign AK certificates.
+	// The TrustedAKs are used first, followed by TrustRoots and Intermediates.
+	// Adding a specific TPM manufacturer's root or intermediate CA means all
+	// TPMs signed by that CA will be trusted.
+	TrustedRoots         *x509.CertPool
+	TrustedIntermediates *x509.CertPool
```

Comment:
Should we just use an `x509.VerifyOptions` structure (https://pkg.go.dev/crypto/x509#VerifyOptions) it already includes Roots and Intermediates? and just override the incompatible fields?

Reply:
That's fine: we'd just have to document that we only take in `Roots` and `Intermediates` from that struct. At that point, it may just be easier to be clear that we only accept those two via top level fields. The only other field of `VerifyOptions` that seems to make sense to expose is `CurrentTime`.

I personally think it's a cleaner abstraction to accept the two fields over a `VerifyOptions` because adding that as a field signals to users of the API that we use other things in `VerifyOptions`

Reply:
Actually I think this is fine. However, we should change the names to `TrustedRootCerts` and `IntermediateCerts` (as the intermediates are not trusted here.